paddle/fluid/operators/conv_cudnn_op.cu

```diff
@@ -68,33 +72,70 @@ class CUDNNConvOpKernel : public framework::OpKernel<T> {
     const std::string data_format = ctx.Attr<std::string>("data_format");
     const bool channel_last = (data_format == "NHWC" || data_format == "NDHWC");
 
+    auto dtype = platform::CudnnDataType<T>::type;
+
+    // Tensor Core introduced from Volta GPUs supports more faster conv op
+    // with FP16 in NHWC data format.
+    const bool compute_in_nhwc =
+        dtype == CUDNN_DATA_HALF && IsVoltaOrLater(dev_ctx);
+    // We will only do data format conversion from NHWC to NCHW.
+    // cudnn will convert NCHW to NHWC automatically on Tensor Core.
+    auto compute_format = compute_in_nhwc && channel_last ? "NHWC" : "NCHW";
+    VLOG(3) << "Compute ConvOp with cuDNN:"
+            << " data_format=" << data_format
+            << " compute_format=" << compute_format;
+
     // ------------ transformed tensor -----------
     Tensor transformed_input_channel(input->type());
     Tensor transformed_output(output->type());
+    Tensor transformed_filter_channel(filter->type());
     T* output_data = nullptr;
-    if (channel_last) {
+    if (channel_last && compute_format == "NCHW") {
+      VLOG(3) << "Transform input tensor from NHWC to NCHW.";
       ResizeToChannelFirst<platform::CUDADeviceContext, T>(
           ctx, input, &transformed_input_channel);
       TransToChannelFirst<platform::CUDADeviceContext, T>(
           ctx, input, &transformed_input_channel);
 
+      ResizeToChannelFirst<platform::CUDADeviceContext, T>(
+          ctx, filter, &transformed_filter_channel);
+      TransToChannelFirst<platform::CUDADeviceContext, T>(
+          ctx, filter, &transformed_filter_channel);
```

Comment:
how to check the layout of the fitler? is it necessary to transform it?

Reply:
I forgot to delete the transform filter layout code after I decided filter layout remains NCHW. Thus filter layout is always NCHW at the begining. This If branch doesn't need to transform filter layout, just as VLOG says. And in the next If branch, we should transform filter branch. I will fix it.

Reply:
OK, it is better to leave a comment about 

> I decided filter layout remains NCHW